src/MainRouter.tsx

```diff
@@ -419,6 +420,21 @@ const MainRouter = (): ReactElement => {
     fetchNotes();
   }, [noteRawTopics]);
 
+  useEffect(() => {
+    const privKey = localStorage.getItem("privKey");
+    const noRedirectedPaths = [
+      ROUTES.WELCOME1,
+      ROUTES.WELCOME2,
+      ROUTES.WELCOME3,
+      ROUTES.WELCOME4,
+      ROUTES.TACONBOARDING,
+      ROUTES.PROFILECREATION,
+    ];
+    if (!privKey && !noRedirectedPaths.includes(location.pathname as ROUTES)) {
+      navigate(ROUTES.APP); // Átirányítás az alap URL-re, ha nincs privKey
```

Comment:
use english

Reply:
maybe add username also

Reply:
A username name is generated at start of the application